main.cpp

```diff
@@ -81,20 +95,37 @@ int main()
         window.draw(sprites["background"]);
         if (game == 1)
         {
+            if (!snake.isPosValid())
+            {
+                looseGame.stop();
+                looseGame.play();
+                game = 0;
+                continue;
+            }
+
             if (clock.getElapsedTime().asMilliseconds() >= SPEED)
             {
                 clock.restart();
                 moveSide = preMoveSide;
                 snake.move(moveSide);
-                if (!snake.inField())
-                {
-                    game = 0;
-                }
             }
         }
+        if (sprites["snakeHead"].getGlobalBounds() == sprites["apple"].getGlobalBounds())
+        {
+            lootApple.stop();
+            lootApple.play();
+            apple.create();
+            snake.increase();
+            if (snake.inApple())
```

Comment:
Suggestion: how about `apple.inSnake(snake)`
Because apple generated in snake, not snake in apple